### ISSUE TITLE
[NavigationView] Setting AutoSuggestBox to x:Null in VisualState doesn't work

### ISSUE THREAD

**Issue by HunterB06** [microsoft/microsoft-ui-xaml#2640]
**Describe the bug**

When trying to hide the `AutoSuggestBox` of a `NavigationView` from a `VisualState`, it throws a COM exception.

**Steps to reproduce the bug**

[NavigationViewIssue.zip](https://github.com/microsoft/microsoft-ui-xaml/files/4759532/NavigationViewIssue.zip)

I linked a simple sample app to test it. It simply has a `NavigationView` containing only the suggest box, and 3 buttons to show what is the behaviour for 3 different scenario:

```xaml
<VisualState x:Name="SuggestBoxChanged">
    <VisualState.Setters>
        <Setter Target="navigationBar.AutoSuggestBox">
            <Setter.Value>
                <AutoSuggestBox PlaceholderText="Changed!" QueryIcon="SolidStar" />
            </Setter.Value>
        </Setter>
    </VisualState.Setters>
</VisualState>
<VisualState x:Name="SuggestBoxNull">
    <VisualState.Setters>
        <Setter Target="navigationBar.AutoSuggestBox" Value="{x:Null}" />
    </VisualState.Setters>
</VisualState>
```

```csharp
private void ChangeSuggestBox_Click(object sender, RoutedEventArgs e) => VisualStateManager.GoToState(this, SuggestBoxChanged.Name, false);

private void RemoveSuggestBox_Click(object sender, RoutedEventArgs e) => navigationBar.AutoSuggestBox = null;

private void RemoveSuggestBoxVisualState_Click(object sender, RoutedEventArgs e) => VisualStateManager.GoToState(this, SuggestBoxNull.Name, false);
```

* We **change** the `AutoSuggestBox` from a `VisualState` => **OK**
* We **remove** the `SuggestBox` from the code behind => **OK**
* We **remove** the `SuggestBox` from a `VisualState` => **FAIL**

**Expected behavior**

What we expect here, is that all scenarios should work. We should be able to hide the suggest box froom the `VisualState`, as we do when we assign it a different `SuggestBox`.

**Version Info**

NuGet package version: 
[Microsoft.UI.Xaml 2.4.2]

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| November 2019 Update (18363)  | Yes   |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | Yes   |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Additional context**

**Comment by HunterB06** [comment]
Yes, if I set its Visibility to Collapsed, the textbox will be hidden but the presenter will still be present (as it appears when the `AutoSuggestBox` property is not null).
```C++
winrt::VisualStateManager::GoToState(*this, box ? L"AutoSuggestBoxVisible" : L"AutoSuggestBoxCollapsed", false /*useTransitions*/);
```

**Comment by JeanRoca** [comment]
Hi @HunterB06 what particular scenario are you looking to drive here? We have `x:Load` which is used for showing and removing content on demand. Would this work in your scenario?

**Comment by HunterB06** [comment]
Hello,

We are looking to make the presenter disappear when the AutoSuggestBox is set to null, which is what it should do, as I detailed in my first message. The fact is that when we set it to null, the UI is not updated properly.
I don't know if x:Load would do the trick, but it should actually work without it. It should be fixed, not workarounded.

**Comment by JeanRoca** [comment]
> Hello,
> 
> We are looking to make the presenter disappear when the AutoSuggestBox is set to null, which is what it should do, as I detailed in my first message. The fact is that when we set it to null, the UI is not updated properly.
> I don't know if x:Load would do the trick, but it should actually work without it. It should be fixed, not workarounded.
> 
> Thanks 🙂,

Thanks for the reply. I understand that this is not a fix for your issue, but in the meantime if this workaround is valid then it is worth highlighting while we track this issue. Unfortunately, we have no ETA for when a fix to this will come.

**Comment by vgromfeld** [comment]
`x:Load` shouldn't do the trick since it will be applied to the `AutoSuggestBox` and would still leave the presenter displayed.

I've made a quick try in my app to see what happens. It throws a `XamlParseException`:

> Failed to assign to property 'Microsoft.UI.Xaml.Controls.NavigationView.AutoSuggestBox'. [Line: 378 Position: 79]

**Comment by evelynwu-msft** [comment]
Repro project using WinAppSDK: [App9.zip](https://github.com/microsoft/microsoft-ui-xaml/files/12009203/App9.zip)

`OptimizedStyle::ResolveValue` needs to be updated to allow `null` to be assigned to properties whose type has the `IsNullable` flag.
